src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const newArr = [];
+
+  for (const action of actions) {
+    const { extraData, keysToRemove, type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(newState, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete newState[key];
+        }
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
```

Comment:
it isn't the best option, because it will be a different object. better to `delete `all properties